From 9bdca9aaf4e6aa1ae61f43bb6be77f4540789374 Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@php.net>
Date: Fri, 12 Oct 2018 13:55:42 +0200
Subject: [PATCH 3/5] Fix [-Wincompatible-pointer-types] with 7.3

---
 src/php7/solr_functions_helpers.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/php7/solr_functions_helpers.c b/src/php7/solr_functions_helpers.c
index 2ac1fe0..eb32808 100644
--- a/src/php7/solr_functions_helpers.c
+++ b/src/php7/solr_functions_helpers.c
@@ -1427,7 +1427,11 @@ static inline int solr_pcre_replace_into_buffer(solr_string_t *buffer, char * se
 {
     zend_string *result;
     int limit = -1;
+#if PHP_VERSION_ID >= 70300
+    size_t replace_count = -1;
+#else
     int replace_count = -1;
+#endif
     zend_string *regex_str = zend_string_init(search, strlen(search), 0);
     zend_string *subject_str = zend_string_init(buffer->str, buffer->len, 0);
 #if PHP_VERSION_ID >= 70200
-- 
2.14.4

